cedar-policy/src/api.rs

```diff
@@ -3662,6 +3662,36 @@ mod policy_set_tests {
             )))
         );
     }
+
+    #[test]
+    fn pset_add_template_conflict_link() {
+        let template = Template::parse(
+            Some("policy0".into()),
+            "permit(principal == ?principal, action, resource);",
+        )
+        .expect("Template Parse Failure");
+        let mut pset = PolicySet::new();
+        pset.add_template(template).unwrap();
+        let env: HashMap<SlotId, EntityUid> =
+            std::iter::once((SlotId::principal(), EntityUid::from_strs("Test", "test"))).collect();
+        pset.link(
+            PolicyId::from_str("policy0").unwrap(),
+            PolicyId::from_str("policy3").unwrap(),
+            env,
+        )
+        .unwrap();
+        let template = Template::parse(
+            Some("policy3".into()),
+            "permit(principal == ?principal, action, resource);",
+        )
+        .expect("Template Parse Failure");
+
+        assert_matches!(
+            pset.add_template(template),
+            Err(PolicySetError::AlreadyDefined)
+        );
+        //Should not panic
```

Comment:
do we already have tests that add a template with the same name as an existing static policy, or as an existing template, throws this error?

Reply:
I didn't add tests for it but from reading the code I'm fairly confident (famous last words)

Reply:
Adding tests for that is probably reasonable just to ensure we don't break it in the future